server/api.go

```diff
@@ -289,6 +295,58 @@ func (p *Plugin) downloadLegalHold(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (p *Plugin) runJobFromAPI(w http.ResponseWriter, _ *http.Request) {
+	_, err := w.Write([]byte("Processing all Legal Holds. Please check the MM server logs for more details."))
+	if err != nil {
+		p.API.LogError("failed to write http response", err.Error())
+	}
+
+	go p.legalHoldJob.RunFromAPI()
+}
+
+// we'll want to store the access secret encrypted in the database
```

Comment:
I put this comment here since right now the UI will be given the stored access secret on page load. Though there is no chance to encrypt/decrypt the config value in the case of saving config values in the system console. Not sure how to proceed with hiding the secret even though it's in plaintext in the server config, and is exposed to the client directly.

Reply:
How is it done in the main mmserver S3 config?

Reply:
Created a ticket to address this https://github.com/mattermost/mattermost-plugin-legal-hold/issues/54